app/models/question.rb

```diff
@@ -1,6 +1,17 @@
 class Question < ActiveRecord::Base
 
   has_many :answers
-  belongs_to :user
+  belongs_to :user, inverse_of: :questions
+
+  def self.all_by_query_params(params)
+    questions = self.where(private: false)
+    if params[:question_name]
+      questions = questions.where('title LIKE ?', '%' + params[:question_name] + '%')
```

Comment:
This is the best way that I've found to do `LIKE` queries in Rails. The second parameter could use interpolation instead of concatenation `"%#{params[:question_name]}%"`, but that's not a big deal.